core/src/main/java/ai/timefold/solver/core/impl/score/stream/bavet/common/DynamicPropagationQueue.java

```diff
@@ -19,15 +19,16 @@
 final class DynamicPropagationQueue<Tuple_ extends AbstractTuple, Carrier_ extends AbstractPropagationMetadataCarrier<Tuple_>>
         implements PropagationQueue<Carrier_> {
 
+    private static final int INITIAL_CAPACITY = 1000; // Selected arbitrarily.
```

Comment:
Would it make sense to use the entity count to estimate capacity?

Reply:
Not really. The tuples could be anything - in case of a groupBy, these will be the groups and therefore the number will likely be very low. In case of a join, the tuples will the the out tuples, and there could be thousands, tens of thousands, even more. IMO there is no good number here.